### PR TITLE
watch etcd for changes

### PR DESCRIPTION
**Pull request by jsravn**
Trigger reconciler sync whenever the etcd store changes. This allows for much quicker update of IPVS state across an IPVS cluster.

### CODE REVIEW

**Comment by jsravn** [comment]
This is ready for review.

File: e2e/meradm/meradm_test.go [hunk 1]
```diff
@@ -133,6 +154,47 @@ var _ = Describe("Meradm", func() {
...
+
+	Describe("External Updates", func() {
+		It("should run reconciler if etcd is updated elsewhere", func() {
+			meradm("service", "add", "service1", "tcp", "10.1.1.1:888", "-s=wrr", "-b=flag-1,flag-2")
+			meradm("server", "add", "service1", "172.16.1.1:555", "-w=1", "-f=masq")
+			time.Sleep(100 * time.Millisecond)
+			MerlinStderr()
+
+			// delete server directly in etcd
+			func() {
```
Comment (alangibson01) [inline e2e/meradm/meradm_test.go]: We could do a little bit of refactoring and remove some of the duplication with the block immediately below.

File: store/store.go [hunk 2]
```diff
@@ -227,3 +231,49 @@ func (s *store) ListServers(ctx context.Context, serviceID string) ([]*types.Rea
...
+	go func() {
+		ctx, cancelFunc := context.WithCancel(context.Background())
+		defer cancelFunc()
+		respCh := make(chan *client.Response)
+		watcherFunc := func() error {
+			resp, err := watcher.Next(ctx)
+			if err == nil {
+				respCh <- resp
+			} else {
+				// stopCh closed
```
Comment (alangibson01) [inline store/store.go]: What does this mean?

Comment (jsravn) [reply store/store.go]: I should rename `closeCh` to `stopCh`. If ctx.Err() != nil, it means something closed the stop channel, and the context got cancelled.

Comment (jsravn) [reply store/store.go]: I'll refactor this and break it up to try and make it clearer.

**Comment by alangibson01** [comment]
Yes, splitting out handleWatcherUpdates does help with readability.

**Pull request merged**